php/commands/site.php

```diff
@@ -161,15 +162,22 @@ public function set_connection_mode($args, $assoc_args) {
   if (in_array($env->get('id'), array('test', 'live'))) {
     $this->failure('Connection mode cannot be set in Test or Live environments');
   }
-  if ($mode == $env->info('connection_mode')) {
-    $this->failure(
-      'The connection mode on {site} for {env} is already set to {mode}.',
-      array(
-        'site' => $site->get('name'),
-        'env' => $env->get('id'),
-        'mode' => $mode
-      ),
-      -1
+  try {
+    $current_mode = $env->info('connection_mode');
+    if ($current_mode == $env->info('connection_mode')) {
+      $this->failure(
+        'The connection mode on {site} for {env} is already set to {mode}.',
+        array(
+          'site' => $site->get('name'),
+          'env' => $env->get('id'),
+          'mode' => $mode
+        ),
+        -1
+      );
+    }
+  } catch (TerminusException $e) {
```

Comment:
Well, I guess we print the logger statement afterwards explaining. So ignore my comment about documenting the exception, and maybe just think about how to isolate these kinds of catches.